### ISSUE TITLE
C code generator with multi output RFC - illegal code generated and general failure to handle multi dimension output

### ISSUE THREAD

**Issue by mg169706** [nok/sklearn-porter#56]
I'm creating a Random Forest Classifier that features 108 outputs. Each output has two labels (on or off). There are two issues I'm having with this:
1. The code generator only seems to create trees for one output. If I have 10 estimators and 10 outputs, I'd expect to see 100 trees in groups of 10 by 10.
2. The code for the single output generates invalid C When I tried to create code for all my outputs. See below for example code fragment.

`int predict_0(float features[]) {
    int classes[[2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 1, 2, 2, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]];
    
    if (features[181] <= 0.5) { ... }
}`

**Comment by MatInGit** [comment]
Any Updates on this issue? I have the same problem.

**Comment by MatInGit** [comment]
Ok, I found a workaround of sorts. You can use `sklearn.multioutput.MultiOutputClassifier` to create a Classifier for each output, then export each `.estimator_` of the multi-output classifier as a separate classifier. It does mean you have to modify the C code a little bit as you now have multiple separate classifiers.

**Comment by HannanKan** [comment]
> Ok, I found a workaround of sorts. You can use `sklearn.multioutput.MultiOutputClassifier` to create a Classifier for each output, then export each `.estimator_` of the multi-output classifier as a separate classifier. It does mean you have to modify the C code a little bit as you now have multiple separate classifiers.

I am afraid it is not a  good idea. 108 outputs will take a lot of labor.
